internal/cluster/certificates.go

```diff
@@ -194,6 +195,17 @@ func (c Certificates) Lookup(ctx context.Context, ctrlclient client.Client, clus
 			return err
 		}
 		certificate.KeyPair = kp
+
+		// Ensure Secret is owned by the Cluster. This provides a migration path for Secrets
+		// previously owned by KubeadmConfigs.
+		patchHelper, err := patch.NewHelper(s, ctrlclient)
+		if err != nil {
+			return err
+		}
+		s.OwnerReferences = certificate.AsSecret(cluster).OwnerReferences
+		if err := patchHelper.Patch(ctx, s); err != nil {
```

Comment:
meh I don't like what I did here. this only updates the ownerReferences on a new join, so conceivably you could have this code deployed and still cascade delete it by deleting the Machine. Should there just be a lookup as part of the core reconciliation path?